azerothcore/azerothcore-wotlk
# Adjust Scholomance Kirtonos boss fight

## Changes Proposed:
- Fixed a bug that was causing Kirtonos to be stuck.
- Adjusted Kirtonos script to more closely match https://youtu.be/nWM5CGoHVZg?t=701 .(kirtonos should switch forms ever 20sec)
  - Added Wing Flap ability in human form according to https://youtu.be/nWM5CGoHVZg?t=701.
  - Removed UNIT_FLAG_NOT_SELECTABLE flag according to https://youtu.be/nWM5CGoHVZg?t=701.
  - Added DoZoneInCombat() to prevent combat reset on targets afflicted with Dominate Mind.


## Issues Addressed:
https://github.com/azerothcore/azerothcore-wotlk/issues/4520
- Closes 1. Kirtonos The Herald is flying in human form, it's extremly weird.
- Bug report in issue comments by @T1ti 



## SOURCE:

Behavior adjustments source: https://youtu.be/nWM5CGoHVZg?t=701
Bug report in issue https://github.com/azerothcore/azerothcore-wotlk/issues/4520 comments by @T1ti 

## Tests Performed:

- Builds without errors.
- Tested in game on linux/docker setup from windows client.

## How to Test the Changes:

##### How to reproduce the Kirtonos stuck bug:
1. Log in and go to Scholomance - Kirtonos room
2. Start the fight with Kirtonos.
3. Deal damage to Kirtonos right after he lands and right before he starts his movement path.
4. Notice if Kirtonos comes to attack but then goes to a spot on the side and is unable to move.
![2021-04-16 16-49-09](https://user-images.githubusercontent.com/10816895/115036854-8a92d280-9ed6-11eb-97f5-930d0d4848a0.gif)

##### Testing the bug fix:
1. Rebuild against the changes in this branch
2. Log in and go to Scholomance - Kirtonos room
3. Start the fight with Kirtonos.
4. Try to reproduce the above bug again. (You should not be able to.)
![2021-04-16 16-35-29](https://user-images.githubusercontent.com/10816895/115037127-d0e83180-9ed6-11eb-89eb-0b3d218134b6.gif)

##### Testing the behavior adjustments:
1. Rebuild against the changes in this branch
2. Log in with 2 characters and go to Scholomance - Kirtonos room.
3. Start the fight with Kirtonos.
4. Check if in human form he is able to do pushback.
5. Check if he changes his form every 20 sec.
6. Check if he casts Dominate Mind only in human form.
7. Check if the party members afflicted by Dominate Mind enter combat after the spell ends.
![2021-04-16 16-41-50](https://user-images.githubusercontent.com/10816895/115037394-160c6380-9ed7-11eb-9e4a-f06f3816acd2.gif)

## Known Issues and TODO List:



## Target Branch(es):







 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
